task_sdk/src/airflow/sdk/definitions/baseoperator.py

```diff
@@ -956,6 +957,25 @@ def __init__(
         self.email_on_retry = email_on_retry
         self.email_on_failure = email_on_failure
 
+        if email is not None:
+            warnings.warn(
+                "email is deprecated please migrate to SmtpNotifier`.",
+                RemovedInAirflow4Warning,
+                stacklevel=2,
+            )
+        if email and email_on_retry is not None:
```

Comment:
Here and the next block, I think you should drop the check for `email`, it's not really relevant.  If they use `email_on_retry` or `email_on_failure` they should get the notice, we don't really care if `email` is set or not at that point, do we? 

Otherwise LGTM assuming the CI finishes green.

Reply:
We must have it because `email_on_retry` has default True and we don't want to change the default.